src/components/Onboarding.js

```diff
@@ -0,0 +1,37 @@
+import { h } from 'preact'
+import { useState, useEffect } from 'preact/hooks'
+import { route } from 'preact-router'
+import { useSoftkey, useI18n } from 'hooks'
+import { onboarding } from 'utils'
+
+export const Onboarding = () => {
+  const i18n = useI18n()
+  const [currentIndex, setCurrentIndex] = useState(0)
+  const softkey = useSoftkey('origin')
```

Comment:
I'm curious why the regular `useSoftkey('origin', { config })` didn't work here

Reply:
oh, it should be `onboarding` not `origin`

`useSoftkey('onboarding', { config })` works, `config` needs to be dynamic when `currentIndex ` change, our current hook doen't allow that

Reply:
`useSoftkey()` has a `dependencies` argument that could contain `currentIndex`. I don't know if it would work or not.

Reply:
it doesn't work, the current solution works only for function call in config `onKeyLeft` `onKeyRight` `onKeyCenter`, `left` `right` are text value that not changing.

Reply:
I'll make it work when I have time.

Reply:
The follow up pr for this is here https://github.com/wikimedia/wikipedia-kaios/pull/92